src/pseudopeople/noise_functions.py

```diff
@@ -149,13 +153,144 @@ def duplicate_with_guardian(
+        & (dataset_data["age"] < 24)
+        & (dataset_data["housing_type"] == "Household")
+        & (dataset_data["guardian_1"].notna())
+    ]
+    in_group_quarters_under_24 = dataset_data.loc[
+        (dataset_data["age"] < 24)
+        & (dataset_data["housing_type"] != "Household")
+        & (dataset_data["guardian_1"].notna())
+    ]
+
+    # Helper function to format group dataframe and merging with their dependentss
+    def _merge_depdents_and_guardians(
+        dependents_df: pd.DataFrame, full_data: pd.DataFrame
+    ) -> pd.DataFrame:
+        # Merge dependents with their guardians. We have to merge twice to check
+        # if either guardian is living at a separate location from the dependent.
+        guardian_1s = full_data.loc[
+            full_data["simulant_id"].isin(full_data["guardian_1"])
+        ].add_prefix("guardian_1_")
+        dependents_and_guardians_df = dependents_df.merge(
+            guardian_1s,
+            how="left",
+            left_on=["guardian_1", "year"],
+            right_on=["guardian_1_simulant_id", "guardian_1_year"],
+        )
+        del guardian_1s
+        guardian_2s = full_data.loc[
+            full_data["simulant_id"].isin(full_data["guardian_2"])
+        ].add_prefix("guardian_2_")
+        dependents_and_guardians_df = dependents_and_guardians_df.merge(
+            guardian_2s,
+            how="left",
+            left_on=["guardian_2", "year"],
+            right_on=["guardian_2_simulant_id", "guardian_2_year"],
+        )
+        del guardian_2s
+        return dependents_and_guardians_df
+
+    # Merge depedents with their guardians
+    formatted_group_data[
+        Keys.ROW_PROBABILITY_IN_HOUSEHOLDS_UNDER_18
+    ] = _merge_depdents_and_guardians(in_households_under_18, dataset_data)
+    formatted_group_data[
+        Keys.ROW_PROBABILITY_IN_HOUSEHOLDS_18_TO_23
+    ] = _merge_depdents_and_guardians(in_households_18_to_23, dataset_data)
+    formatted_group_data[
+        Keys.ROW_PROBABILITY_IN_GROUP_QUARTERS_UNDER_24
+    ] = _merge_depdents_and_guardians(in_group_quarters_under_24, dataset_data)
+    # Note: We have two dicts (configuration and formatted_group_data) at this point that have
+    # the key for the group and then a dataframe for that group or the group and the configured
+    # noise level
+
+    noised_data = []
+    for group, group_df in formatted_group_data.items():
+        # Get index groups that can be noised based on dependent and guardian(s) addresses
+        both_different_index = group_df.index[
+            (group_df["household_id"] != group_df["guardian_1_household_id"])
+            & (group_df["guardian_1_household_id"].notna())
+            & (group_df["household_id"] != group_df["guardian_2_household_id"])
+            & (group_df["guardian_2_household_id"].notna())
+        ]
+        # Choose which guardian to copy when dependent lives in different address from both guardains
+        choices = randomness_stream.choice(
```

Comment:
What is this doing w/o defined probabilities? Uniform?

Reply:
If a dependent lives in a different address than both their guardians (say someone in college who has 2 guardians that live in a house). We can only duplicate each dependent once so we have to pick which guardian will duplicate the dependent. This way we will only have a maximum of 2 rows for a dependent, 1 row for their actually address and 1 row for either guardian 1 or guardian 2.

Reply:
Right but you're not passing in the probabilities of each choice getting chosen. I assume it defaults to 50/50?

Reply:
Yes random choice defaults to 1/len(options) if you don't pass it probabilities.